pep-9999.rst

```diff
@@ -0,0 +1,520 @@
+releases every other month, as is being proposed for the rolling beta releases.
+
+
+Design Discussion
+=================
+
+Why this proposal over simply doing more frequent baseline feature releases?
+----------------------------------------------------------------------------
+
+For large parts of Python's user base, *availability* of new CPython feature
+releases isn't the limiting factor on their adoption of those new releases.
+As such, any proposal based on speeding up full feature releases needs to strike
+a balance between meeting the needs of users who would be adopting each release
+as it became available, and those that would now be in a position of adopting
+every 2nd, 3rd, or 4th release, rather than migrating to every release (albeit
+potentially with a lag of 12 months or more).
+
+This proposal aims to approach the problem from a different angle by defining a
+*new* production-ready release stream that is more specifically tailored to the
+interests of operating environments that are in a position to consume new
+releases as fast as the CPython core team is prepared to produce them.
+
+
+Why "continuous beta" rather than "continuous alpha"?
+-----------------------------------------------------
+
+The code quality standards upheld by the CPython code review process and
+BuildBot fleet make the "beta" label more suitable than the "alpha" label.
+
+The "production ready, but subject to change" caveat also aligns well with the
+"beta" term.
+
+
+Why not use Calendar Versioning for the rolling release stream?
+---------------------------------------------------------------
+
+Steve Dower's initial write-up of this proposal [1_] suggested the use of
+calendar versioning for the rolling release stream (so Python 2019.12 rather
+than 3.9.0b1, etc).
+
+Paul Moore pointed out [2_] two major practical problems with that proposal:
+
+* it makes it hard for users of the calendar-based versions to know where they
+  stand in relation to the traditionally numbered versions
+* it breaks ``Python-Requires`` metadata processing (since all calendar
+  versions would appear as newer than any standard version)
+
+This PEP aims to address both of those problems by using the established beta
+version numbers for the rolling releases.
+
+The beta numbering approach also avoids other questions raised by the calendar
+versioning concept, such as how ``sys.version_info``, ``Py_VERSION_HEX``,
+``site-packages`` directory naming, and installed binary naming would work.
+
+
+Implications for CPython core development
+-----------------------------------------
+
+The major change for CPython core development is the need to keep the master
+branch more consistently release ready.
+
+While the main requirement for that would be to keep the stable BuildBot fleet
+green, there would also be encouragement to keep the development version of
+the documentation up to date for the benefit of users of the rolling beta
+releases.
```

Comment:
We should also reach a better consensus on patterns for implementing feature detection and flags within CPython and the standard library. I don't think we have to define them all here, but it is worth acknowledging that we expect/hope to see an increase in:
* features that are deliberately detectable through conditional imports/hasattr rather than version checks
* features that are `from __future__` enabled within a beta release, that may become always-on while still in beta

Reply:
The reasoning for these being that explicit detection/enabling like this makes it trivial for users of the beta release stream to notice when we remove or change provisional features (`from __future__` breaks on compile), or to safely fall back on previous functionality (we can add warnings for imports/attribute checks that we can't for version checks)

Reply:
Liberally lifted from these comments for a new design discussion section.